hermes/bin/src/pool.rs

```diff
@@ -0,0 +1,97 @@
+//! Global thread pool for parallel WASM module execution.
+//!
+//! This module initializes a global Rayon thread pool
+//! for running WASM modules in parallel. The pool uses
+//! work-stealing for efficient load balancing across
+//! available threads. To avoid saturating the system,
+//! the pool leaves a small margin of CPU capacity
+//! unused — typically 2 threads are reserved for the
+//! main application flow and event loop.
+//!
+//! Each event may execute multiple WASM modules
+//! concurrently within this pool.
+
+use std::{
+    sync::{atomic::AtomicUsize, Condvar, Mutex},
+    thread::available_parallelism,
+};
+
+use anyhow::{Context, Result};
+use rayon::ThreadPoolBuilder;
+
+/// Global counter of currently running tasks.
+static TASK_COUNTER: AtomicUsize = AtomicUsize::new(0);
```

Comment:
Why is it a usize?  What relationship does it have to pointers or memory?  Is it used for indexing?
Unless its got something to do with memory (an index or pointer math) we should pick an appropriately sized variable type so its 100% consistent regardless of platform.
Given that its actually counting concurrently running threads, it probably never needs to be bigger than u16.